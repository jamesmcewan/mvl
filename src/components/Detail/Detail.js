import React from 'react';
import styled from 'styled-components';
import Creators from '../Creators/Creators';

const DetailWrapper = styled.div`
  padding-bottom: 26px;
  margin: 0 13px;

  @media screen and (min-width: 700px) {
    padding: 0;
    margin: 0 6px 6px;
  }

  @media screen and (min-width: 1000px) {
    display: flex;
    flex-flow: row wrap;
    align-items: flex-start;
    margin-left: 52px;
  }
`;

const DetailHeading = styled.h2`
  font-family: 'Roboto Slab', serif;
  margin-top: 0;
  @media screen and (min-width: 1000px) {
    flex: 1 100%;
  }
`;

const DiamondCode = styled.p`
  @media screen and (min-width: 1000px) {
    flex: 1 100%;
  }
`;

const DetailDescription = styled.div`
  margin-bottom: 26px;

  @media screen and (min-width: 1000px) {
    max-width: 600px;
    margin-right: 52px;
  }
`;

const DetailButton = styled.button`
  border: 1px solid #ff79c6;
  background: transparent;
  font-size: 18px;
  padding: 9px;
  border-radius: 10px;
  color: #ff79c6;
`;

const Detail = ({ title, description, creators, diamondCode, urls, close }) => (
  <DetailWrapper>
    <DetailHeading>{title}</DetailHeading>
    <DetailDescription dangerouslySetInnerHTML={{ __html: description }} />
    <Creators creators={creators} />
    <DiamondCode>{diamondCode}</DiamondCode>
    <DetailButton onClick={e => close(e)}>Close</DetailButton>
  </DetailWrapper>
);

Detail.defaultProps = {
  title: '',
  description: '',
  creators: [
    {
      name: '',
      role: '',
    },
  ],
  dates: [
    {
      date: '',
      type: '',
    },
  ],
  urls: [
    {
      url: '',
      type: '',
    },
  ],
};

export default Detail;
