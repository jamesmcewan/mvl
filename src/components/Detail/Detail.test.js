import React from 'react';
import { mount } from 'enzyme';
import Detail from './Detail';
import 'jest-styled-components';

describe('Detail', () => {
  it('should render without crashing', () => {
    const wrapper = mount(<Detail />);
    expect(wrapper).toHaveLength(1);
  });

  it('should render the correct JSX', () => {
    const wrapper = mount(<Detail />);
    expect(wrapper).toMatchSnapshot();
  });
});
