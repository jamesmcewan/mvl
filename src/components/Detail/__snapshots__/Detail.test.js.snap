// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail should render the correct JSX 1`] = `
.c4 {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

.c3 {
  margin-top: 0;
  font-size: 22px;
}

.c0 {
  padding-bottom: 26px;
  margin: 0 13px;
}

.c1 {
  font-family: 'Roboto Slab',serif;
  margin-top: 0;
}

.c2 {
  margin-bottom: 26px;
}

.c6 {
  border: 1px solid #ff79c6;
  background: transparent;
  font-size: 18px;
  padding: 9px;
  border-radius: 10px;
  color: #ff79c6;
}

@media screen and (min-width:700px) {
  .c0 {
    padding: 0;
    margin: 0 6px 6px;
  }
}

@media screen and (min-width:1000px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-flow: row wrap;
    -ms-flex-flow: row wrap;
    flex-flow: row wrap;
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
    margin-left: 52px;
  }
}

@media screen and (min-width:1000px) {
  .c1 {
    -webkit-flex: 1 100%;
    -ms-flex: 1 100%;
    flex: 1 100%;
  }
}

@media screen and (min-width:1000px) {
  .c5 {
    -webkit-flex: 1 100%;
    -ms-flex: 1 100%;
    flex: 1 100%;
  }
}

@media screen and (min-width:1000px) {
  .c2 {
    max-width: 600px;
    margin-right: 52px;
  }
}

<Detail
  creators={
    Array [
      Object {
        "name": "",
        "role": "",
      },
    ]
  }
  dates={
    Array [
      Object {
        "date": "",
        "type": "",
      },
    ]
  }
  description=""
  title=""
  urls={
    Array [
      Object {
        "type": "",
        "url": "",
      },
    ]
  }
>
  <styled.div>
    <div
      className="c0"
    >
      <styled.h2>
        <h2
          className="c1"
        />
      </styled.h2>
      <styled.div
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
      >
        <div
          className="c2"
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
        />
      </styled.div>
      <Creators
        creators={
          Array [
            Object {
              "name": "",
              "role": "",
            },
          ]
        }
      >
        <div>
          <styled.h3>
            <h3
              className="c3"
            >
              Creative team
            </h3>
          </styled.h3>
          <styled.ul>
            <ul
              className="c4"
            >
              <Creator
                key="c-"
                name=""
                role=""
              >
                <li>
                   - 
                </li>
              </Creator>
            </ul>
          </styled.ul>
        </div>
      </Creators>
      <styled.p>
        <p
          className="c5"
        />
      </styled.p>
      <styled.button
        onClick={[Function]}
      >
        <button
          className="c6"
          onClick={[Function]}
        >
          Close
        </button>
      </styled.button>
    </div>
  </styled.div>
</Detail>
`;
